assets/app/view/game/corporate_sell_shares.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'view/game/actionable'
+require 'view/game/sell_shares'
+
+module View
+  module Game
+    class CorporateSellShares < Snabberb::Component
+      include Actionable
+
+      def render
+        @step = @game.round.active_step
+        current_actions = @step.current_actions
+        @entity ||= @game.current_entity
+        children = []
+
+        children.concat(render_sources) if current_actions.include?('corporate_sell_shares')
+
+        h('div.margined', children.compact)
```

Comment:
there's no reason to compact, you will never have nils, maybe you can try

h('div.margined', render_sources)

def render_sources
  return [] unless current_actions.include?